conda/cli/main_env_update.py

```diff
@@ -143,22 +144,20 @@ def execute(args: Namespace, parser: ArgumentParser) -> int:
             raise CondaError(
                 dals(
                     f"""
-                    Unable to install package for {0}.
+                    Unable to install package for {installer_type}.
 
-                    Please double check and ensure you dependencies file has
-                    the correct spelling.  You might also try installing the
-                    conda-env-{0} package to see if provides the required
-                    installer.
+                    Please double check and ensure your dependencies file has
+                    the correct spelling. You might also try installing the
+                    corresponding conda plugin for this type (e.g. conda-pypi
+                    for 'pip').
```

Comment:
I'm not sure we should hardcode particular plugins in the error message. The plugin nuance is good to have, but I wouldn't mention conda-pypi. If anything, we could link to the [conda/plugins](https://github.com/conda-incubator/plugins) repo (which we haven't really used too much), but that can be done separately.